assignments/assignment02.rb

```diff
@@ -73,3 +107,156 @@ def pluck(ary, key)
 # - daily balance
 # - summary:
 #   - starting balance, total deposits, total withdrawals, ending balance
+
+
+def create_transaction(date, payee, amount, type)
+  {date: date, payee: payee, amount: amount, type: type}
+end
+
+File.open("bankdata.csv") do |input|
+  ary = input.readlines
+  header = ary.shift
+
+  records = ary.map do |line|
+    fields = line.split ','
+    create_transaction fields[0], fields[1], fields[2], fields[3].chop
+  end
+
+  def extract_withdrawal(records)
+    withdrawals = []
+    records.each do |line|
+      if line[:type] == 'withdrawal'
+        withdrawals << line
+      end
+    end
+    withdrawals
+  end
+
+  def extract_deposit(records)
+    deposits = []
+    records.each do |line|
+      if line[:type] == 'deposit'
+        deposits << line
+      end
+    end
+    deposits
+  end
+
```

Comment:
notice that extract_withdrawal and extract_deposit are basically the same, except for the type. This means you have repeated code. You could have created a single method that takes the type as a parameter.